test/shared/src/test/scala/zio/test/TestUtils.scala

```diff
@@ -12,6 +14,18 @@ object TestUtils {
         (p, fail(s))
     }.recover { case _ => (false, fail(s)) }
 
+  def nonFlaky[R, E](test: ZIO[R, E, Boolean]): ZIO[R, E, Boolean] = {
```

Comment:
Implementation could be: `test.repeat(Schedule.recurs(100) *> Schedule.identity[Boolean])`

Reply:
Thanks! Much nicer! Now if we can only get rid of that pesky `Clock` type constraint.

Reply:
Which we can shortly, I hope, with #1348!